src/gitlab.js

```diff
@@ -115,25 +126,35 @@ export default class VerdaccioGitLab implements IPluginAuth {
       // In legacy mode, the groups are:
       // - for access, depending on the package settings in verdaccio
       // - for publish, the logged in user id and all the groups they can reach as fixed `$auth.gitlab.publish` = `$owner`
+      const gitlabAccessQueryParams = this.config.legacy_mode ? { owned: true } : { min_access_level: accessLevelId };
```

Comment:
Duplicate the logic for getting groups based on the **access** level for the user (contrary to the **publish** level).